wagtail/admin/widgets/chooser.py

```diff
@@ -166,3 +167,23 @@ def media(self):
             versioned_static('wagtailadmin/js/page-chooser-modal.js'),
             versioned_static('wagtailadmin/js/page-chooser.js'),
         ])
+
+
+class PageChooserAdapter(Adapter):
+    js_constructor = 'wagtail.widgets.PageChooser'
+
+    def js_args(self, widget, context):
+        return [
+            widget.render_html('__NAME__', None, attrs={'id': '__ID__'}),
+            widget.id_for_label('__ID__'),
+            widget.client_config,
+        ]
+
+    @property
+    def media(self):
+        return AdminPageChooser().media + forms.Media(js=[
```

Comment:
Is the order here significant? If `AdminPageChooser` depends on `widgets.js`, should `widgets.js` be added first so it's loaded before `AdminPageChooser`'s js?

Reply:
I'd say the dependency is the other way round (i.e. the current ordering is correct): AdminPageChooser's media contains the base implementation of the widget that comes into play regardless of whether we instantiated it through telepath or oldschool server-side rendering (with an inline `<script>createPageChooser(...)</script>` call), and telepath/widgets.js is the wrapper around it to make it interoperate with telepath.

(It just happens that the internal `chooser` object from createPageChooser is exactly what telepath needs, and so that wrapper is more or less transparent.)